WordPress/gutenberg
# BaseControl: Add `box-sizing` resets

Part of #42415 
Part of #41961

## What?

- Add a `box-sizing` reset to BaseControl. This immediately fixes the box-sizing problems we were seeing in the following components:
  - TextControl: Wrong width
  - TextareaControl: Wrong width
  - ToggleGroupControl: Wrong height
- Removes the ad hoc `box-sizing` declarations in components based on BaseControl.

## Why?

We can't guarantee that the surrounding context has a box-sizing reset in place.

## Testing Instructions

1. `npm run storybook:dev`
2. See that the problems with TextControl, TextareaControl, and ToggleGroupControl have been fixed.
3. Extra credit: In your IDE, search for `<BaseControl` to see where it's used in the repo, especially in packages other than wp-components. Take a look at the code (or test in wp-env!) to see if there could be problems.

✅ I have visually inspected all the BaseControl-based components compared with `trunk` to check for regressions
